### PR TITLE
[MDEP-789] Improve documentation of analyze - Non-test scoped

### PR DESCRIPTION
**Pull request by slawekjaranowski**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MDEP) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MDEP-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MDEP-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java [hunk 1]
```diff
@@ -167,7 +169,7 @@
 
     /**
      * List of dependencies that will be ignored. Any dependency on this list will be excluded from the "declared but
-     * unused" and the "used but undeclared" list. The filter syntax is:
+     * unused", the "used but undeclared" and the "non-test scoped" list. The filter syntax is:
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java:172]: comma after undeclared"

File: src/site/apt/examples/exclude-dependencies-from-dependency-analysis.apt.vm [hunk 2]
```diff
@@ -34,10 +34,11 @@ Exclude dependencies from dependency analysis
   that contain annotations and the byte code analysis is unable to
   determine whether a jar is actually required or not.
 
-  The plugin can then be configured to ignore these dependencies in
-  either "declared but unused" or "undeclared but used" case or in
-  both simultaneously.  See the following POM configuration for an
-  example:
+  The plugin can then be configured to ignore dependencies in
+  "declared but unused", "undeclared but used" and "non-test scoped"
```
Comment (elharo) [inline src/site/apt/examples/exclude-dependencies-from-dependency-analysis.apt.vm:38]: comma after used"

File: src/site/apt/examples/exclude-dependencies-from-dependency-analysis.apt.vm [hunk 3]
```diff
@@ -34,10 +34,11 @@ Exclude dependencies from dependency analysis
   that contain annotations and the byte code analysis is unable to
   determine whether a jar is actually required or not.
 
-  The plugin can then be configured to ignore these dependencies in
-  either "declared but unused" or "undeclared but used" case or in
-  both simultaneously.  See the following POM configuration for an
-  example:
+  The plugin can then be configured to ignore dependencies in
```
Comment (elharo) [inline src/site/apt/examples/exclude-dependencies-from-dependency-analysis.apt.vm:37]: in --> that are

File: src/site/apt/examples/exclude-dependencies-from-dependency-analysis.apt.vm [hunk 4]
```diff
@@ -57,7 +58,7 @@ Exclude dependencies from dependency analysis
             <configuration>
               <failOnWarning>true</failOnWarning>
 
-              <!-- ignore jsr305 for both "used but undeclared" and "declared but unused" -->
+              <!-- ignore jsr305 for "used but undeclared", "declared but unused" and "non-test scoped" -->
```
Comment (elharo) [inline src/site/apt/examples/exclude-dependencies-from-dependency-analysis.apt.vm:61]: comma after unused"

**Comment by slawekjaranowski** [comment]
fixed

**Pull request merged**